packages/pages/src/dev/server/ssr/generateTestData.ts

```diff
@@ -62,6 +70,31 @@ export const generateTestDataForSlug = async (
   return getDocumentByLocale(parsedData, locale);
 };
 
+/**
+ * In dev mode, we cannot use the loadTemplateModules that uses esbuild.
+ * Trying to do so results in losing loader support for .ico files and possibly
+ * other file types.
+ * Instead we have to rely on vite to import templates.
+ */
+const loadTemplateModuleCollectionUsingVite = async (
```

Comment:
I'm confused why ico files are relevant to generateTestData. The only thing we're trying to do is load the modules in order to get the config off of them to pass to generateTestData.

Reply:
I feel like this is fine for now since this matches what we were doing before - if we want to update serverRenderSlugRoute to use ts-morph or loadTemplateModules we'd want serverRenderRoute to also match which I think is out of scope here

Reply:
Fair enough